app/Http/Controllers/Dashboard/ApiController.php

```diff
@@ -89,4 +110,18 @@ public function getIncidentTemplate()
 
         throw new ModelNotFoundException("Incident template for $templateSlug could not be found.");
     }
+
+    /**
+     * Checks if Cachet is up to date.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function checkVersion()
```

Comment:
Also, for performance reasons, we can resolve as late as possible, so we might bail out before we need to resolve an object.